OsmAnd/src/net/osmand/plus/exploreplaces/ExplorePlacesFragment.kt

```diff
@@ -51,41 +103,111 @@ class ExplorePlacesFragment : BaseOsmAndFragment(), NearbyPlacesAdapter.NearbyIt
 		dialogFragment?.hide()
+		bottomSheetBehavior = BottomSheetBehavior.from(mainContent!!)
+		bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
+		bottomSheetBehavior.peekHeight =
+			resources.getDimensionPixelSize(R.dimen.bottom_sheet_menu_peek_height)
+		bottomSheetBehavior.isHideable = true
+		bottomSheetBehavior.addBottomSheetCallback(object :
+			BottomSheetBehavior.BottomSheetCallback() {
+			override fun onStateChanged(bottomSheet: View, newState: Int) {
+				AndroidUiHelper.updateVisibility(
+					showOnMapContainer,
+					newState == BottomSheetBehavior.STATE_EXPANDED && !isMapVisible)
+				when (newState) {
+					BottomSheetBehavior.STATE_EXPANDED -> {
+						showListContainer?.visibility = View.GONE
+					}
+
+					BottomSheetBehavior.STATE_COLLAPSED -> {
+						showListContainer?.visibility = View.GONE
+					}
+
+					BottomSheetBehavior.STATE_HIDDEN -> {
+						showListContainer?.visibility = View.VISIBLE
+					}
+
+					BottomSheetBehavior.STATE_SETTLING -> {
+						showListContainer?.visibility = View.GONE
+					}
+				}
+			}
+
+			override fun onSlide(bottomSheet: View, slideOffset: Float) {
+			}
+		})
 	}
 
 	fun onBackPress(): Boolean {
-		if (isHidden) {
+		if (isMapVisible) {
 			if (mapActivity?.contextMenu?.isVisible == true) {
 				mapActivity?.contextMenu?.hideMenus()
+			} else if (bottomSheetBehavior.state != BottomSheetBehavior.STATE_HIDDEN) {
+				hideList()
 			} else {
-				activity?.supportFragmentManager?.beginTransaction()
-					?.show(this@ExplorePlacesFragment)
-					?.commit()
+				isMapVisible = false
+				mainContent?.visibility = View.VISIBLE
+				bottomSheetBehavior.state = BottomSheetBehavior.STATE_EXPANDED
 			}
-			return true
 		} else {
 			val quickSearchFragment = mapActivity?.fragmentsHelper?.quickSearchDialogFragment
 			quickSearchFragment?.show()
 			activity?.supportFragmentManager?.beginTransaction()
 				?.remove(this@ExplorePlacesFragment)
 				?.commit()
-			return true
 		}
+		return true
 	}
 
-	private fun setupShowAll(view: View) {
+	override fun onResume() {
+		super.onResume()
+		app.locationProvider.addLocationListener(this)
+		app.locationProvider.addCompassListener(this)
+		mapActivity?.let { activity -> updateWidgetsVisibility(activity, View.GONE) }
+	}
 
+	override fun onPause() {
+		super.onPause()
+		val app = requireActivity().application as OsmandApplication
+		app.locationProvider.removeLocationListener(this)
+		app.locationProvider.removeCompassListener(this)
+		mapActivity?.let { activity -> updateWidgetsVisibility(activity, View.VISIBLE) }
+	}
+
+	override fun updateLocation(location: Location?) {
+		this.location = location
+		verticalNearbyAdapter.updateLocation(location)
+	}
+
+	override fun updateCompassValue(value: Float) {
+		val now = System.currentTimeMillis()
+		if (now - lastCompassUpdate > COMPASS_UPDATE_PERIOD) {
+			lastCompassUpdate = now
+			updateLocation(location)
+		}
+	}
```

Comment:
It is better to add checks for changes to reduce unnecessary list updates, [example](https://github.com/osmandapp/Osmand/blob/25319b1aad25ae880a699a3e969708fab7887bb0/OsmAnd/src/net/osmand/plus/myplaces/favorites/dialogs/FavoritesTreeFragment.java#L242-L256)